src/main/java/com/jakewharton/retrofit2/adapter/kotlin/coroutines/experimental/CoroutineCallAdapterFactory.kt

```diff
@@ -99,7 +99,13 @@ class CoroutineCallAdapterFactory private constructor() : CallAdapter.Factory()
 
         override fun onResponse(call: Call<T>, response: Response<T>) {
           if (response.isSuccessful) {
-            deferred.complete(response.body()!!)
```

Comment:
Can't we use the built in `checkNotNull()` function?

Reply:
Thank you!
I was forgotten `checkNotNull()` function. :sweat:
However, when using `checkNotNull()` function, you need to cast `response.body` whose type is` T?`To` Any?`.
You also need to revert to `deferred.complete (body!!)`.

Is this an appropriate usage?

```kt
val body = response.body()
checkNotNull(body as Any?) {
  deferred.completeExceptionally(NullPointerException("Response body is null"))
  return
}

deferred.complete(body!!)
````

Reply:
`checkNotNull` is not appropriate here. It's equivalent to `!!`

Reply:
`let` can be use instead `!!` 
 i.e
```response.body()?.let { deferred.complete(it) }```

Reply:
No. That completely changes the behavior and hides errors.